pastas/check.py

```diff
@@ -0,0 +1,555 @@
+    parameters: Optional[Union[list[str], str]] = None,
+    n_std: float = 2.0,
+):
+    """Check if parameter value is larger than n_std times the standard deviation.
+
+    Note that it is the modelers responsibility to check if the estimated uncertainty
+    is reliable!
+
+    Parameters
+    ----------
+    ml: pastas.Model
+        Pastas model instance.
+    parameters: list or str, optional
+        List of parameter names to check the uncertainty for. Default is None, which
+        means all parameters are checked.
+    n_std: float, optional
+        Number of standard deviations to compare the parameter to. Default is 2.
+
+    """
+    if parameters is None:
+        parameters = ml.parameters.index.tolist()
+    elif isinstance(parameters, str):
+        parameters = [parameters]
+
+    # loop through parameters
+    results = []
+    for parameter in parameters:
+        results.append(_uncertainty_parameter(ml, parameter, n_std=n_std))
+    return concat(results)
+
+
+def _uncertainty_parameter(ml, parameter, n_std=2):
+    """Internal method to check if parameter value is larger than n_std * std.
+
+    Parameters
+    ----------
+    ml: pastas.Model
+        Pastas model instance.
+    parameter: str
+        Name of the parameter to check.
+    n_std: float, optional
+        Number of standard deviations to compare the parameter to. Default is 2.
+
+    Returns
+    -------
+    df: pandas.DataFrame
+        DataFrame with the results of the check.
+    """
+    df = DataFrame(
+        columns=["statistic", "operator", "threshold", "unit", "pass", "comment"]
+    )
+    df.index.name = "check"
+    # get stressmodel
+    sm_name = "_".join(parameter.split("_")[:-1])
+    if sm_name in ml.stressmodels:
+        sm = ml.stressmodels[sm_name]
+    else:
+        sm = None
+    # WellModel gain is a special case, because the parameter depends on distance
+    if sm is not None and sm._name == "WellModel" and parameter.endswith("_A"):
+        nwells = sm.distances.index.size
+        for iw in range(nwells):
+            params = sm.get_parameters(model=ml, istress=iw)
+            p = sm.rfunc.gain(params)
+            std = sm.variance_gain(model=ml, istress=iw)
+            check = np.abs(p) > (n_std * std)
+            df.loc[f"|{parameter} ({sm.distances.index[iw]})| > {n_std}σ"] = [
+                p,
+                ">",
+                n_std * std,
+                guess_unit(parameter),
+                check,
+                "Assumes estimate of σ is reliable.",
+            ]
+    else:
+        p = ml.parameters.loc[parameter, "optimal"]
+        std = ml.parameters.loc[parameter, "stderr"]
+        check = np.abs(p) > (n_std * std)
+        df.loc[f"|{parameter}| > {n_std}σ"] = [
+            p,
+            ">",
+            n_std * std,
+            guess_unit(parameter),
+            check,
+            "Assumes estimate of σ is reliable.",
+        ]
+    return df
+
+
+def guess_unit(parameter):
```

Comment:
This function returns both units and dimensions. So maybe we should only give dimensions. Because units are undoable I think.

Reply:
For some cases we definitely know the units, so it felt weird not to return them, but I suppose going for just dimensions is more consistent.

Reply:
I've switched it to returning dimensions, but sort of keeping support for both. I renamed the unit column to dimensions. Once others take a look we can determine what we want to do with this...